Source/OxyPlot.Avalonia.Shared/PlotBase.cs

```diff
@@ -0,0 +1,250 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="PlotBase.cs" company="OxyPlot">
+//   Copyright (c) 2014 OxyPlot contributors
+// </copyright>
+// <summary>
+//   Represents a control that displays a <see cref="PlotModel" />.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+using Avalonia.Reactive;
+
+namespace OxyPlot.Avalonia
+{
+    using global::Avalonia;
+    using global::Avalonia.Controls;
+    using global::Avalonia.Controls.Primitives;
+    using global::Avalonia.Input;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+    using System.Threading;
+
+    /// <summary>
+    /// Represents a control that displays a <see cref="PlotModel" />.
+    /// </summary>
+    public abstract partial class PlotBase : TemplatedControl, IPlotView
+    {
+        private Control currentTracker;
+        protected Panel panel;
+        private Canvas overlays;
+        private ContentControl zoomControl;
+        private ScreenPoint mouseDownPoint;
+
+        protected const string PartPanel = "PART_Panel";
+
+        /// <summary>
+        /// Invalidation flag (0: no update, 1: update, 2: update date).
+        /// </summary>
+        protected internal int isUpdateRequired;
+
+        /// <summary>
+        /// Invalidation flag (0: no update, 1: update visual elements).
+        /// </summary>
+        protected internal int isRenderRequired;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PlotBase" /> class.
+        /// </summary>
+        protected PlotBase()
+        {
+            this.TrackerDefinitions = [];
```

Comment:
This is really going to annoy me: hard not to see this as an immutable list. I assume these are automatic fixes? (I'd prefer the full type was here, given it's opaque; `new()` would be preferable to `[]` otherwise, but this is just personal preference)

Reply:
Yes those are automated fixes which are offered on latest Visual Studio on default settings. I don't mind this notation and don't have the association with immutable lists, but I can see how this might be an opinionated matter.

There are a couple of issues regarding code style in this PR. The problem is that there are neither clear guidelines (even less so than in the core repo) nor a .editorconfig which could be used to make sure the style is ok.

Reply:
Yeah, I just need to get with the times; it's fine. Could pull the .editorconfig from the core repo, that would be fine by me